server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClient.java

```diff
@@ -39,6 +39,11 @@ public interface CoordinatorClient
    */
   ListenableFuture<DataSegment> fetchUsedSegment(String dataSource, String segmentId);
 
+  /**
+   * Fetches segment metadata for the given dataSource and segmentId.
+   */
+  ListenableFuture<DataSegment> fetchSegment(String dataSource, String segmentId);
```

Comment:
I made it a separate API instead of a parameter since this probably shouldn't be used in most other places, so adding an extra parameter for every current call seemed untidy. I am okay with changing it to a parameter since it would cut down on the interface though.

Reply:
Since there are 3 places in production code where you would have to make the changes. I guess it should be fine?

 
<img width="1721" alt="Screenshot 2023-08-18 at 10 21 20 AM" src="https://github.com/apache/druid/assets/2260045/527e4bc0-fe5a-4bb0-b210-ba4034f52599">